src/rad/appdata.cpp

```diff
@@ -1380,15 +1373,14 @@ bool ApplicationData::LoadProject( const wxString &file, bool justGenerate )
 	return true;
 }
 
-bool ApplicationData::ConvertProject( const wxString& path, int fileMajor, int fileMinor )
+bool ApplicationData::ConvertProject(ticpp::Document& doc, const wxString& path, int fileMajor, int fileMinor)
 {
 	try
 	{
 		ticpp::Document doc;
 		XMLUtils::LoadXMLFile( doc, false, path );
```

Comment:
This can not and does not work (i actually tested this). The method parameter doc gets shadowed by the local one, your conversion gets lost at the end of the method.

Removing the shadowing is not enough, the calling code stores a reference to the root node of the document (Line 1275), but this method actually erases the document and rebuilds it so the reference of the outer code will break. This needs also to be addressed.